### PR TITLE
DMP-1857 Modified case search to handle global and courthouse judge

### PR DESCRIPTION
**Pull request by johnrace0**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-1857

### Change description ###
Case Search - Handle Global & Courthouse Judge

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java [hunk 1]
```diff
@@ -71,12 +71,12 @@ private List<Predicate> createPredicates(GetCasesSearchRequest request, Criteria
         List<Predicate> predicates = new ArrayList<>();
         CollectionUtils.addAll(predicates, createCaseCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addHearingDateCriteria(request, criteriaBuilder, caseRoot));
-        CollectionUtils.addAll(predicates, addCourthouseCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addCourtroomCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addJudgeCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addDefendantCriteria(request, criteriaBuilder, caseRoot));
         CollectionUtils.addAll(predicates, addEventCriteria(request, criteriaBuilder, caseRoot));
         if (!userIdentity.userHasGlobalAccess(Set.of(JUDGE))) {
+            CollectionUtils.addAll(predicates, addCourthouseCriteria(request, criteriaBuilder, caseRoot));
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java:79]: This doesn't look right to me - if you look at `addCourthouseCriteria` that just seems to be adding the extra search criteria for the courthouse field.  It isn't doing anything with the security tables we have for the user's courthouses via Table: darts.security_group_courthouse_ae

I'd say we need that to be restored and the user security conditions changed here.

File: src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java [hunk 2]
```diff
@@ -71,12 +71,12 @@ private List<Predicate> createPredicates(GetCasesSearchRequest request, Criteria
...
             CollectionUtils.addAll(predicates, addUserSecurityRolesCriteria(criteriaBuilder, caseRoot));
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java:80]: The `addUserSecurityRolesCriteria` method maybe needs to be renamed as it doesn't seem to be looking at security roles, but security groups to determine the user's courthouses via the `joinUser` method?

Comment (johnrace0) [reply src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java:80]: renamed - I don't know if the original was joined by roles - but changed to groups now

**Comment by johnrace0** [comment]
Closed - current implementation is working ok

**Pull request closed without merging**